### PR TITLE
Add Devise as an Add-on

### PR DESCRIPTION
**Pull request by noor-ahmed**
## What happened
Solved https://github.com/nimblehq/rails-templates/issues/164
_Devise is an awesome authentication solution for Rails projects. But sometimes devs prefer own solution. So moving it as an option to this template_
 
## Insight

- [x] Added template file
- [x] Added Devise and Devise I18n gem(commented initially)
- [x] Moved Specs from base to under templates
- [x] Added devise support for spec

## Proof Of Work
Screen Shot 2020-10-09 at 6 03 26 PM

### CODE REVIEW

File: template.rb [hunk 1]
```diff
@@ -45,14 +45,23 @@ def apply_template!(template_root)
   apply 'config/template.rb'
   apply '.gitignore.rb'
 
+  if ENV['CI']
```
Comment (andyduong1920) [inline template.rb:48]: we don't need the ENV['CI'] anymore @noor-ahmed, you can check how another addon did that for example https://github.com/nimblehq/rails-templates/pull/232/files

File: template.rb [hunk 2]
```diff
@@ -45,14 +45,23 @@ def apply_template!(template_root)
...
+    apply '.template/addons/devise/template.rb'
+  elsif yes?("Would you like to add Devise gem?")
+    @add_devise = true
+    apply '.template/addons/devise/template.rb'
+  end
+
   after_bundle do
     use_source_path template_root
 
+    generate 'devise:install' if @add_devise
```
Comment (andyduong1920) [inline template.rb:59]: can we move this one to `devise` addon itself?

File: template.rb (hunk 1 above)
Comment (noor-ahmed) [reply template.rb:48]: Yep, we'll remove it. We kept it for time being only.

File: template.rb [hunk 3]
```diff
@@ -45,14 +45,23 @@ def apply_template!(template_root)
   apply 'config/template.rb'
   apply '.gitignore.rb'
 
+  if ENV['CI']
+    @add_devise = true
+    apply '.template/addons/devise/template.rb'
+  elsif yes?("Would you like to add Devise gem?")
```
Comment (rafayet-monon) [inline template.rb:51]: I think we can use `install_addon_prompt` now after rebasing with `develop`

Comment (noor-ahmed) [reply template.rb]: Yeah, already done cc7b7c1 ✌️

File: template.rb (hunk 2 above)
Comment (noor-ahmed) [reply template.rb:51]: Moved to` devise/template.rb` https://github.com/nimblehq/rails-templates/pull/250/commits/069359d26a95bb75d8b1aadab4cea6553423a6f2

**Review by rafayet-monon — COMMENTED** [review]
I think we need to edit the [Makefile](https://github.com/nimblehq/rails-templates/blob/develop/Makefile). Add another `Y\n`

```
# Y - in response to Would you like to add PhraseApp configuration?
# Y - in response to Would you like to add Devise addon?

create:
  printf "Y\nY\n" | rails new $(APP_NAME) -m ./template.rb -T ${OPTIONS}
```

File: Makefile [hunk 4]
```diff
@@ -1,4 +1,6 @@
 # Y - in response to Would you like to add PhraseApp configuration?
+# Y - in response to Would you like to add Devise addon?
+
 create:
 	printf "Y\n" | rails new $(APP_NAME) -m ./template.rb -T ${OPTIONS}
```
Comment (rafayet-monon) [inline Makefile:5]: Missed `Y\n` 😅 
```suggestion
  printf "Y\nY\n" | rails new $(APP_NAME) -m ./template.rb -T ${OPTIONS}
```

**Review by rafayet-monon — APPROVED** [review]
LGTM 🎉

File: template.rb (hunk 2 above)
Comment (andyduong1920) [reply template.rb:51]: 🤔  it's still here @noor-ahmed something wrong?

File: README.md [hunk 5]
```diff
@@ -46,6 +46,7 @@ Available Addons:
 - `docker`
 - `phrase_app`
 - `semaphore`
+- [`devise`](https://github.com/heartcombo/devise)
```
Comment (andyduong1920) [inline README.md:49]: Let keep the format as normal text without the link same like another addon, for now, @noor-ahmed as the user know what's devise before using it I guess.

File: Makefile [hunk 6]
```diff
@@ -1,6 +1,8 @@
 # Y - in response to Would you like to add PhraseApp configuration?
+# Y - in response to Would you like to add Devise addon?
+
 create:
-	printf "Y\n" | rails new $(APP_NAME) -m ./template.rb -T ${OPTIONS}
+	printf "Y\nY\n" | rails new $(APP_NAME) -m ./template.rb -T ${OPTIONS}
```
Comment (andyduong1920) [inline Makefile:5]: please rebase `develop` @noor-ahmed

File: template.rb [hunk 7]
```diff
@@ -100,6 +105,10 @@ def delete_test_folder
   FileUtils.rm_rf('test')
 end
 
+def install_addon_prompt(addon)
```
[the remark this answers is not in the document]
Comment (noor-ahmed) [reply template.rb]: It's already outdated, I already resolved it :'D

File: Makefile (hunk 6 above)
Comment (noor-ahmed) [reply Makefile]: rebase! 👍

**Review by andyduong1920 — APPROVED** [review]
LGTM @noor-ahmed

**Pull request merged**